pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go

```diff
@@ -121,6 +121,15 @@ func (c *EtcdEndpointsController) syncConfigMap(ctx context.Context, recorder ev
 	endpointAddresses := make(map[string]string, len(members))
 	// Create endpoint addresses for each member of the cluster.
 	for _, member := range members {
+		// Since a learner member rejects all requests other than serializable reads and member status API
+		// we have to exclude them from the etcd-endpoints configmap to prevent the API server from sending
+		// requests to it which would fail with "rpc not supported for learner" errors
+		// TODO(haseeb): Exclude learners from MemberList() everywhere else where it might recevie an unsupported request
+		// e.g defrag controller.
+		if member.IsLearner {
+			continue
+		}
+
 		if member.Name == "etcd-bootstrap" {
```

Comment:
we should also check if a machine for a member is not pending deletion. In that case, the member should not be added. This will prevent reading previously deleted members.

Reply:
This check already exists in the clustermember controller. So any machine with a deletion timestamp would not get re-added to the membership, and as a consequence will never end up in the etcd-endpoints configmap.